Plugin/test1/src/main/java/util/InstrumentationUtil.java

```diff
@@ -0,0 +1,135 @@
+package util;
+
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.*;
+import com.intellij.psi.search.FilenameIndex;
+import com.intellij.psi.search.GlobalSearchScope;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * An class containing common utility methods to simplify the instrumentation actions
+ */
+public final class InstrumentationUtil {
+
+    /**
+     * Scan the project directory for all source files to search for all Java source files
+     *
+     * @param project An object representing an IntelliJ project.
+     * @return A list of all Java source files in the project
+     */
+    public static @NotNull List<PsiFile> getAllJavaFilesInProjectAsPsi(Project project) {
+        List<PsiFile> psiFiles = new LinkedList<>();
+        String[] fileNames = FilenameIndex.getAllFilenames(project);
+
+        fileNames = Arrays.stream(fileNames)
+                .filter(fileName -> fileName.contains(".java"))
+                .toArray(String[]::new);
+
+        for (String fileName : fileNames) {
+            PsiFile[] psiJavaFiles = FilenameIndex.getFilesByName(project, fileName, GlobalSearchScope.projectScope(project));
+            // Remove the files from the NAPPA library from the list to process
+            psiJavaFiles = Arrays.stream(psiJavaFiles)
+                    .filter(psiJavaFile -> !((PsiJavaFile) psiJavaFile).getPackageName().contains("nl.vu.cs.s2group"))
```

Comment:
This excludes the NAPPA files from the instrument search.
To allow the usage of NAPPA in any Android project developed by the S2 Group, we would need to encapsulate the NAPPA library in the `nl.vu.cs.s2group.NAPPA` package instead.

Reply:
Created issue #29 for this.